client/src/components/Common/ListHeader.vue

```diff
@@ -1,25 +1,31 @@
 <script setup lang="ts">
-import { library } from "@fortawesome/fontawesome-svg-core";
 import { faAngleDown, faAngleUp, faBars, faGripVertical } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
-import { BButton } from "bootstrap-vue";
+import { BButton, BButtonGroup, BFormCheckbox } from "bootstrap-vue";
 import { computed, ref } from "vue";
 
+import { defaultSortKeys, type ListView, type SortBy, type SortKey } from "@/components/Common";
 import { useUserStore } from "@/stores/userStore";
 
-library.add(faAngleDown, faAngleUp, faBars, faGripVertical);
-
-type ListView = "grid" | "list";
-type SortBy = "create_time" | "update_time" | "name";
-
 interface Props {
+    sortKeys?: SortKey[];
```

Comment:
Please sort props and function names alphabetically throughout. It makes navigating the code easier.